annotation-processor/src/main/java/org/hibernate/validator/ap/checks/annotationparameters/AnnotationParametersGroupsCheck.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks.annotationparameters;
+
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.type.DeclaredType;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckError;
+import org.hibernate.validator.ap.util.AnnotationApiHelper;
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+/**
+ * Checks that the groups parameter of any constraint annotation contains only interfaces.
+ *
+ * @author Marko Bekhta
+ */
+public class AnnotationParametersGroupsCheck extends AnnotationParametersAbstractCheck {
+
+
+	public AnnotationParametersGroupsCheck(AnnotationApiHelper annotationApiHelper) {
+		super( annotationApiHelper );
+	}
+
+
+	@Override
+	protected boolean canCheckThisAnnotation(AnnotationMirror annotation) {
+		return true;
+	}
+
+	@Override
+	protected Set<ConstraintCheckError> doCheck(Element element, AnnotationMirror annotation) {
+		List<? extends AnnotationValue> annotationValue = annotationApiHelper.getAnnotationArrayValue( annotation, "groups" );
+		Set<ConstraintCheckError> issues = CollectionHelper.newHashSet();
+
+		for ( AnnotationValue value : annotationValue ) {
+			if ( value.getValue() instanceof DeclaredType && !( (DeclaredType) value.getValue() ).asElement().getKind().isInterface() ) {
```

Comment:
hmm... this `instanceof` is more for safety of next conversion. I've added it more for the cases when the annotation is of some different kind and groups parameter is present but contains something else ... But if it's only for constraint annotations it can be probably be removed completely. WDYT ?